test/unit/text/shaping.cpp

```diff
@@ -65,39 +65,39 @@ TEST_CASE("shaping")
     mapnik::face_manager fm(fl, font_file_mapping, font_memory_cache);
 
     {
-        std::vector<std::tuple<unsigned, unsigned>> expected =
-            {{{0, 0}, {0, 3}, {0, 4}, {0, 7}, {3, 8}, {11, 9}, {68, 10}, {69, 11}, {70, 12}, {12, 13}}};
+        std::vector<std::pair<unsigned, unsigned>> expected =
+            {{0, 0}, {0, 3}, {0, 4}, {0, 7}, {3, 8}, {11, 9}, {68, 10}, {69, 11}, {70, 12}, {12, 13}};
         // with default NotoSans-Regular.ttc and NotoNaskhArabic-Regular.ttf ^^^
```

Comment:
@artemp the reason this works with pair but not tuple is because pair is an aggregate, tuple isn't (because it has a base). If you ever need to do this with more than a pair of values, or another non-aggregate-type, the trick is to specify the type of the first element; subsequent elements don't need the type repeated, so: 
```c++
std::vector<std::tuple<unsigned, unsigned>> expected = {std::make_tuple(0u, 0u), {0, 3}, {0, 4}};
// and they don't even need to be put in a vector,
// you can just capture the `std::initializer_list<T>` by reference
auto & expected = {std::tuple<unsigned, unsigned>{0, 0}, {0, 3}, {0, 4}};
```